src/reporter/coverage-report.xsl

```diff
@@ -431,8 +431,10 @@
          <xsl:for-each select="$hits">
             <xsl:variable name="hit-traceable-uqname" as="xs:string?"
                select="exactly-one(key('traceables', @traceableId, $trace))/@uqname" />
+            <xsl:variable name="hit-traceable-class" as="xs:string?"
+               select="exactly-one(key('traceables', @traceableId, $trace))/@class" />
             <xsl:if test="($node-uqname eq $hit-traceable-uqname) or
-                          empty($hit-traceable-uqname)">
+                          exists($hit-traceable-class)">
                <xsl:sequence select="." />
             </xsl:if>
          </xsl:for-each>
```

Comment:
I am pretty sure this block of code can be simpler, maybe even removing the conditional logic. But it might be useful to do such refactoring in a follow-up pull request to make it easier to see whether any of the expected HTML coverage reports in `test/end-to-end/cases-coverage/` actually change as a result of simplifying the XSLT.